tools/exporter/csvExporter.py

```diff
@@ -0,0 +1,349 @@
+        self.items = audit_data_and_items[1]
+        self.auditdata_array = self.retrieve_auditdata()
+        self.data = self.process_items()
+
+    def process_items(self):
+        """
+        convert json of audit items to 2 dimensional list
+
+        :return:    2 diminsional list, each list is a single item, which corresponds to a single row
+        """
+        self.data = []
+        for item in self.items:
+            item_array = self.generate_csv_row_item_data(item)
+            row_array = item_array + self.auditdata_array
+            self.data.append(row_array)
+        return self.data
+
+    def write(self, path, filename, write_or_append='wb'):
+        """
+        write audit data to csv, this can be run immediately after instance of class is initialized
+
+        :param path:            path to desired output location
+        :param filename:        desired name of file
+        :param write_or_append:
+        :return:
+        """
+        if filename[-4:] != '.csv':
+            filename = filename + '.csv'
+        file_path = os.path.join(path, filename + '.csv')
+        if not os.path.isfile(file_path) or write_or_append == 'wb':
+            self.data.insert(0, CSV_HEADER_ROW)
+        try:
+            csv_file = open(file_path, write_or_append)
+            wr = csv.writer(csv_file, quoting=csv.QUOTE_ALL)
+            for row in self.data:
+                wr.writerow(row)
+            csv_file.close()
+        except Exception as ex:
+            print str(ex) + ': Exception while writing' + filename + ' to file'
+
+    def path(self, obj, *args):
+        """
+        Check for the existence of path for a given JSON object
+        :param obj:     JSON Object
+        :param args:    target path, list of keys
+        :return:        if path exists, return value, otherwise return empty string
+        """
+        for arg in args:
+            if isinstance(obj, list) and isinstance(arg, int):
+                if len(obj) == 0:
+                    return ''
+                obj = obj[arg]
+            elif arg in obj.keys():
+                obj = obj[arg]
+            else:
+                return ''
+        return obj
+
+    # noinspection PyDictCreation
+    def generate_csv_row_item_data(self, item):
+        """
+        Process item JSON data into array that CSV writer can handle
+        :param item:    single item in JSON format
+        :return:        array of item data, in format that CSV writer can handle
+        """
+
+        fields = {
+            COMMENTS: BLANK,
+            FAILED: BLANK,
+            SCORE: BLANK,
+            MAX_SCORE: BLANK,
+            SCORE_PERCENTAGE: BLANK,
+            PARENT_ID: BLANK,
+            RESPONSE: BLANK
+        }
+
+        # all fields have a parent id, grab this and insert into fields dictionary
+        fields[PARENT_ID] = self.path(item, 'parent_id')
+        # if failed key exists, insert value into dictionary
```

Comment:
removed comment as there is not an unclear 'why?' question regarding this line of code